`popf` package - different behaviour of cmake outputs in nix develop and nix build

I'm trying to build humble's `popf` package.

When i run dev environment `nix develop github:lopsided98/nix-ros-overlay?ref=staging#humble.popf` and run in sequence `cmake .` and `make` commands within unpacked release tarball it is successfully configuring and builds.

But when i build package via `nix build github:lopsided98/nix-ros-overlay?ref=staging#humble.popf` configurePhase finished successful, but with CMake warnings:
<details><summary>CMake  warnings in configurePhase</summary>

    CMake Warning: Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025
    PYTHON_EXECUTABLE
    PYTHON_INCLUDE_DIR
    PYTHON_LIBRARY
</details>

and buildPhase finished with an error:
```
... bash: line 1: /build/popf-release-release-humble-popf-0.0.14-1/src/VALfiles/parsing/fixyywrap: cannot execute: required file not found
``` 

Im not sure, but maybe in can be caused by custom cmake command? (https://github.com/fmrico/popf/blob/humble-devel/CMakeLists.txt#L59):
```
add_custom_command(
       OUTPUT
        ${CMAKE_CURRENT_BINARY_DIR}/pddl+.cpp
       COMMAND
        ${BISON_EXECUTABLE}
       ARGS
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/pddl+.yacc -o ${CMAKE_CURRENT_BINARY_DIR}/pddl+.cpp
       DEPENDS
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/pddl+.yacc ${CMAKE_CURRENT_BINARY_DIR}/lex.yy.cc
       COMMAND
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/
)
```

@wentasah Thank you very much. I'm applied this patch and successfully built `popf` package.

While preparing pull request to nix-ros-overlay with your patch (it should be in `distros/distro-overlay.nix`) i additionally tried to build whole `plansys2-bringup` package and stuck with `plansys2-executor`.

In first time it throws Cmake configure error:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ZMQ_INCLUDE_DIR
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1
   used as include directory in directory /build/ros2_planning_system-release-release-humble-plansys2_executor-2.0.9-1

CMake Error in CMakeLists.txt:
  Found relative path while evaluating include directories of
  "plansys2_executor":

    "ZMQ_INCLUDE_DIR-NOTFOUND"

CMake Error in CMakeLists.txt:
  Found relative path while evaluating include directories of
  "executor_node":

    "ZMQ_INCLUDE_DIR-NOTFOUND"
```
After adding zeromq in `nativeBuildInputs` in overlay configurePhase was done, but in buildPhase i got:
```
> [100%] Linking CXX executable executor_node
> /nix/store/fhzz4yrdy17czwc9i4swhlpcp445inzb-binutils-2.40/bin/ld: libplansys2_executor.so: undefined reference to `BT::PublisherZMQ::PublisherZMQ(BT::Tree const&, unsigned int, unsigned int, unsigned int)'
> collect2: error: ld returned 1 exit status
> make[2]: *** [CMakeFiles/executor_node.dir/build.make:347: executor_node] Error 1
> make[1]: *** [CMakeFiles/Makefile2:165: CMakeFiles/executor_node.dir/all] Error 2
> make: *** [Makefile:146: all] Error 2
```

Are you tried to build entire plansys2 for humble?

--- a/distros/distro-overlay.nix
+++ b/distros/distro-overlay.nix
@@ -171,6 +171,15 @@ let
       nativeBuildInputs = nativeBuildInputs ++ [ self.qt5.wrapQtAppsHook ];
     });
 
+    popf = rosSuper.popf.overrideAttrs ({
+      nativeBuildInputs ? [], ...
+    }: {
+      nativeBuildInputs = nativeBuildInputs ++ [ self.perl ];
+      postPatch = ''
+        patchShebangs --build src/VALfiles/parsing/fixyywrap
+      '';
+    });
+
     pr2-tilt-laser-interface = patchBoostSignals rosSuper.pr2-tilt-laser-interface;
 
     python-qt-binding = rosSuper.python-qt-binding.overrideAttrs ({

--- a/distros/distro-overlay.nix
+++ b/distros/distro-overlay.nix
@@ -172,10 +172,10 @@ let
     });
 
     popf = rosSuper.popf.overrideAttrs ({
-      nativeBuildInputs ? [], ...
+      nativeBuildInputs ? [], postPatch ? "", ...
     }: {
       nativeBuildInputs = nativeBuildInputs ++ [ self.perl ];
-      postPatch = ''
+      postPatch = postPatch + ''
         patchShebangs --build src/VALfiles/parsing/fixyywrap
       '';
     });

